kong/cli/utils/signal.lua

```diff
@@ -62,8 +64,16 @@ local NGINX_SEARCH_PATHS = {
 -- Try to find an `nginx` executable in defined paths, or in $PATH
 -- @return Path to found executable or nil if none was found
 local function find_nginx()
-  for i = 1, #NGINX_SEARCH_PATHS + 1 do
-    local prefix = NGINX_SEARCH_PATHS[i] and NGINX_SEARCH_PATHS[i] or ""
```

Comment:
But iirc, this was taken from Lapis. 

And on that particular line we have the or condition where the prefix is an empty string, which is supposed to search in the PATH already if standards paths failed. 

Manually go through each part of the PATH seems like the job of the shell, not of Kong. We should rather investigate if it works, and why it it would eventually doesn't. 

> On Oct 12, 2015, at 5:20 PM, Marco Palladino <notifications@github.com> wrote:
> 
> In kong/cli/utils/signal.lua:
> 
> > @@ -62,8 +64,16 @@ local NGINX_SEARCH_PATHS = {
> >  -- Try to find an `nginx` executable in defined paths, or in $PATH
> >  -- @return Path to found executable or nil if none was found
> >  local function find_nginx()
> > -  for i = 1, #NGINX_SEARCH_PATHS + 1 do
> > -    local prefix = NGINX_SEARCH_PATHS[i] and NGINX_SEARCH_PATHS[i] or ""
> Because there is an issue where the nginx executable is in a different folder specified into the PATH environment variable, and we are not checking that. If you read the line below, now this will check in both our paths and the PATH environment variable.
> 
> —
> Reply to this email directly or view it on GitHub.
>